src/alerts/version-polling.ts

```diff
@@ -1,20 +1,19 @@
-import { getConfig } from 'src/libs/config';
 import { withErrorIgnoring } from 'src/libs/error';
 
-import { getBadVersions, getLatestVersion, latestVersionStore } from './version-alerts';
+import { getBadVersions, getLatestVersion, versionStore } from './version-alerts';
 
 export const VERSION_POLLING_INTERVAL = 15 * 60 * 1000; // 15 minutes
 
 export const checkVersion = withErrorIgnoring(async (): Promise<void> => {
-  const currentVersion = getConfig().gitRevision;
+  const { currentVersion } = versionStore.get();
 
   const latestVersion = await getLatestVersion();
-  latestVersionStore.set(latestVersion);
+  versionStore.update((value) => ({ ...value, latestVersion }));
 
   if (latestVersion !== currentVersion) {
     const badVersions = await getBadVersions();
     if (badVersions.includes(currentVersion)) {
-      window.location.reload();
+      versionStore.update((value) => ({ ...value, isUpdateRequired: true }));
```

Comment:
Do we ever need to reset `isUpdateRequired` back to false after the user had a bad version and then their current version changes to a non-bad version?

Reply:
@nawatts nudge on this when you get a chance

Reply:
Ah, it looks like the subsequent PR changed this from a boolean to a timestamp. https://github.com/DataBiosphere/terra-ui/pull/4559

Reply:
Currently, this is a one way thing: the page automatically refreshes before it polls for bad versions again. I think it's also unlikely that we mark a version "bad" and then undo that within a few minutes. And even if we did do that, I think it's a less confusing UX to go ahead and refresh unnecessarily than show a countdown to refresh and then cancel it and do nothing.

Reply:
When the page is reloaded, all the application state is re-initialized and `isUpdateRequired` is initialized to false.